es-space-mgmt/main/default/lwc/reservationDetailForm/__tests__/reservationDetailForm.test.js

```diff
@@ -0,0 +1,244 @@
+            );
+            const lighntingComboBoxEl = element.shadowRoot.querySelector(
+                'lightning-combobox'
+            );
+            const lighntingSliderEl = element.shadowRoot.querySelector(
+                'lightning-slider'
+            );
+            const lighntingInputEl = element.shadowRoot.querySelector(
+                'lightning-input'
+            );
+            const lighntingRadioGroupEl = element.shadowRoot.querySelector(
+                'lightning-radio-group'
+            );
+            expect(lighntingLayoutEl).not.toBeNull();
+            expect(lighntingComboBoxEl).not.toBeNull();
+            expect(lighntingComboBoxEl.options).toStrictEqual(CITY_OPTIONS);
+            expect(lighntingComboBoxEl.placeholder).toBe('Choose a market');
+            expect(lighntingSliderEl).not.toBeNull();
+            expect(lighntingInputEl).not.toBeNull();
+            expect(lighntingRadioGroupEl).not.toBeNull();
+            expect(lighntingRadioGroupEl.options).toMatchObject(DATE_RANGES);
+        });
+    });
+
+    it('getMarketsByState @wire return no data', () => {
+        const element = createElement('c-reservation-detail-form', {
+            is: ReservationDetailForm
+        });
+        element.state = STATE;
+        element.city = CITY;
+        element.people = 50;
+        element.startdate = START_DATE;
+        element.totaldays = 15;
+        document.body.appendChild(element);
+        // Emit mock response with 0 rows
+        getMarketsByStateAdapter.emit(mockMarketsByStateNoRecords);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            const lighntingLayoutEl = element.shadowRoot.querySelector(
+                'lightning-layout'
+            );
+            const lighntingComboBoxEl = element.shadowRoot.querySelector(
+                'lightning-combobox'
+            );
+            const lighntingSliderEl = element.shadowRoot.querySelector(
+                'lightning-slider'
+            );
+            const lighntingInputEl = element.shadowRoot.querySelector(
+                'lightning-input'
+            );
+            const lighntingRadioGroupEl = element.shadowRoot.querySelector(
+                'lightning-radio-group'
+            );
+            expect(lighntingLayoutEl).not.toBeNull();
+            expect(lighntingComboBoxEl).not.toBeNull();
+            expect(lighntingComboBoxEl.options).toStrictEqual([]);
+            expect(lighntingComboBoxEl.placeholder).toBe('No markets found');
+            expect(lighntingSliderEl).not.toBeNull();
+            expect(lighntingInputEl).not.toBeNull();
+            expect(lighntingRadioGroupEl).not.toBeNull();
+        });
+    });
+
+    it('change input handlers', () => {
```

Comment:
This test can be removed. In the next test you are testing the draft submission, and that is sufficient (as that would fail if any of the handlers would not do their thing).

Basically copy/paste all the dispatch stuff into the below draft test.

Reply:
I have made changes as needed for other tests. I did not want to mix the event dispatch for the draft use case and the change handlers for other inputs. I think its worth keeping them separate!